### PR TITLE
fix(v2): make not clickable post title on post item page

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

On the post page, the title should not be clickable

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

image

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-theme-classic/src/theme/BlogPostItem/index.js [hunk 1]
```diff
@@ -45,7 +45,12 @@ function BlogPostItem(props) {
     return (
       <header>
         <h1 className={classnames('margin-bottom--sm', styles.blogPostTitle)}>
-          <Link to={permalink}>{title}</Link>
+          {/* eslint-disable no-restricted-globals */}
+          {location.pathname !== permalink ? (
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/theme/BlogPostItem/index.js]: Is there a better solution?

File: packages/docusaurus-theme-classic/src/theme/BlogPostItem/index.js [hunk 2]
```diff
@@ -45,7 +51,7 @@ function BlogPostItem(props) {
     return (
       <header>
         <h1 className={classnames('margin-bottom--sm', styles.blogPostTitle)}>
-          <Link to={permalink}>{title}</Link>
+          {isBlogPostPage ? title : <Link to={permalink}>{title}</Link>}
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/theme/BlogPostItem/index.js]: Is there a better solution? Comparison with `location.pathname` will not work, what other options are there?

**Comment by endiliey** [comment]
https://endiliey.com/

**Comment by lex111** [comment]
> Also, not all blog post has “Read more”. If there is no “<!—truncate” marker in the markdown, it wont have read more button

But in posts lists, the title will also remain clickable, isn't it? This change will only affect the post item page.

**Comment by endiliey** [comment]
oops you are right. I misread it on mobile :(

File: packages/docusaurus-theme-classic/src/theme/BlogPostItem/index.js (hunk 2 above)
Comment (endiliey) [reply packages/docusaurus-theme-classic/src/theme/BlogPostItem/index.js]: I think this is the right solution (?). Determining whether it's a list or not using pathname parsing can lead to subtle bugs

**Comment by lex111** [comment]
Is there anything else to do before it can be merged?

**Comment by endiliey** [comment]
I think this is OK. I find it annoying too previously (the blog list page and item page had no different so i keep on clicking the same thing).

I'm not UI expert so I usually ask for @yangshun opinion on this. but LGTM

File: packages/docusaurus-theme-classic/src/theme/BlogPostPage/index.js [hunk 3]
```diff
@@ -20,7 +20,10 @@ function BlogPostPage(props) {
         <div className="container margin-vert--xl">
           <div className="row">
             <div className="col col--8 col--offset-2">
-              <BlogPostItem frontMatter={frontMatter} metadata={metadata}>
+              <BlogPostItem
+                frontMatter={frontMatter}
+                metadata={metadata}
+                isBlogPostPage>
```
Comment (yangshun) [inline packages/docusaurus-theme-classic/src/theme/BlogPostPage/index.js]: Prefer to do `isBlogPostPage={true}`

Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/BlogPostPage/index.js]: Hmm.. OK, fixed :heavy_check_mark:

File: packages/docusaurus-theme-classic/src/theme/BlogPostItem/index.js (hunk 2 above)
Comment (yangshun) [reply packages/docusaurus-theme-classic/src/theme/BlogPostItem/index.js]: I agree this is a right way to check.

File: packages/docusaurus-theme-classic/src/theme/BlogPostPage/index.js (hunk 3 above)
Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/BlogPostPage/index.js]: @yangshun 

CI fails:
```
  26:17  error  Value must be omitted for boolean attributes  react/jsx-boolean-value
```

What have I done wrong?

Comment (yangshun) [reply packages/docusaurus-theme-classic/src/theme/BlogPostPage/index.js]: I think our lint rules prefers not explicitly writing the boolean value 🤦‍♂ Sorry let's change it back.

Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/BlogPostPage/index.js]: Fixed! :+1:

**Pull request merged**